backend/editor/api.py

```diff
@@ -35,4 +74,152 @@ async def pong(response: Response):
     Check server health
     """
     pong = datetime.now()
-    return {"ping": "pong @ %s" % pong}
\ No newline at end of file
+    return {"ping": "pong @ %s" % pong}
+
+@app.get("/nodes")
+async def findAllNodes(response: Response):
+    """
+    Get all nodes within taxonomy
+    """
+    result = get_all_nodes("")
+    allNodes = list(result)
+    return allNodes
+
+@app.get("/entry/{entry}")
+async def findOneEntry(response: Response, entry: str):
+    """
+    Get entry corresponding to id within taxonomy
+    """
+    result = get_nodes("ENTRY", entry)
+    oneEntry = list(result)
+
+    check_single(oneEntry)
+    
+    return oneEntry[0]
+
+@app.get("/entry")
+async def findAllEntries(response: Response):
+    """
+    Get all entries within taxonomy
+    """
+    result = get_all_nodes("ENTRY")
+    allEntries = list(result)
+    return allEntries
+
+@app.get("/synonym/{synonym}")
+async def findOneSynonym(response: Response, synonym: str):
+    """
+    Get synonym corresponding to id within taxonomy
+    """
+    result = get_nodes("SYNONYMS", synonym)
+    oneSynonym = list(result)
+
+    check_single(oneSynonym)
+
+    return oneSynonym[0]
+
+@app.get("/synonym")
+async def findAllSynonyms(response: Response):
+    """
+    Get all synonyms within taxonomy
+    """
+    result = get_all_nodes("SYNONYMS")
+    allSyononyms = list(result)
+    return allSyononyms
+
+@app.get("/stopword/{stopword}")
+async def findOneStopword(response: Response, stopword: str):
+    """
+    Get stopword corresponding to id within taxonomy
+    """
+    result = get_nodes("STOPWORDS", stopword)
+    oneStopword = list(result)
+    
+    check_single(oneStopword)
+
+    return oneStopword[0]
+
+@app.get("/stopword")
+async def findAllStopwords(response: Response):
+    """
+    Get all stopwords within taxonomy
+    """
+    result = get_all_nodes("STOPWORDS")
+    allStopwords = list(result)
+    return allStopwords
+
+@app.get("/header")
+async def findHeader(response: Response):
+    """
+    Get __header__ within taxonomy
+    """
+    result = get_nodes("TEXT", "__header__")
+    header = list(result)
+    return header[0]
+
+@app.get("/footer")
+async def findFooter(response: Response):
+    """
+    Get __footer__ within taxonomy
+    """
+    result = get_nodes("TEXT", "__footer__")
+    footer = list(result)
+    return footer[0]
+
+# Post methods
+
+@app.post("/edit/entry/{entry}")

```

Comment:
Sorry for I didn't check it.

Usually on APIs the best is to just have 

```suggestion
@app.post("/entry/{entry}")
```

It's a common design pattern, that with a GET you have the entry, with a POST you modify it.

I think it's worth it, if you agree @aadarsh-ram

Reply:
Yep, I agree. I'll make the changes